rest/context.go

```diff
@@ -130,12 +133,13 @@ type RequestContext interface {
 type gorillaRequestContext struct {
 	context.Context
 	req      *http.Request
+	router   *mux.Router
 	messages []string
 }
 
 // NewContext returns a RequestContext populated with parameters from the request path and
 // query string.
-func NewContext(parent context.Context, req *http.Request) RequestContext {
+func NewContext(parent context.Context, req *http.Request, router *mux.Router) RequestContext {
```

Comment:
I thought this would be relatively safe as it isn't directly set by the user within handlers. I do see a lot of calls in gotable-server tests though.

We could always add a `NewContextWithRouter` or something and preserve the original.